strings/string_is_valid_number.py

```diff
@@ -0,0 +1,134 @@
+"""
+Solution By: Reniz Shah
+Topic: Deterministic Finite Automaton (DFA)
+Given a string s, return whether s is a valid number or not
+Input: s = -90E3
+Output: True
```

Comment:
Is this input–output pair tested anywhere? Why leave it at the top of the file?

Reply:
Yes this pair is tested you can check at line129 in new PR
>>> is_valid_number("-90E3")
    True

Reply:
So then there's no need to have this test at the top of the file, so let's remove it

Reply:
Ok, I removed it